src/test/components/credentialsForm/credentialsForm.test.js

```diff
@@ -193,57 +150,38 @@ describe('CredentialsForm', () => {
 
     actions.addMessage = jest.fn().mockImplementation(() => ({ type: 'nonsense' }));
 
-    await act(async () => {
-      wrapper = mount(
-        componentWrapperIntl(
-          <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
-          store,
-          initialEntry
-        )
-      );
-    });
-    wrapper.update();
+    render(
+      componentWrapperIntl(
+        <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
+        store,
+        initialEntry
+      )
+    );
 
-    await act(async () => {
-      wrapper.find('input[name="authentication.username"]').instance().value = 'newname';
-      wrapper.find('input[name="authentication.username"]').simulate('change');
-    });
-    wrapper.update();
+    await waitFor(() => expect(() => screen.getByRole('progressbar')).toThrow());
+
+    userEvent.type(screen.getAllByRole('textbox')[0], '{selectall}{backspace}newname');
 
     expect(updateAuthentication).not.toHaveBeenCalled();
     expect(actions.addMessage).not.toHaveBeenCalled();
 
-    jest.useFakeTimers();
-
-    await act(async () => {
-      wrapper.find('form').simulate('submit');
-    });
-    wrapper.update();
+    userEvent.click(screen.getByText('Submit'));
 
-    expect(wrapper.find(MemoryRouter).instance().history.location.pathname).toEqual(
-      replaceRouteId(routes.sourcesDetail.path, sourceId)
-    );
+    expect(screen.getByTestId('location-display').textContent).toEqual(replaceRouteId(routes.sourcesDetail.path, sourceId));
```

Comment:
Same as above, no need to check for the `textContent` you can use directly option's `name` attribute.